### DISCUSSION TITLE
Diff, apply, and prune on multiple inline envs

### DISCUSSION THREAD

**Discussion by curusarn** [grafana/tanka#656] [Ideas]
I have noticed that it is possible to `export` multiple tanka envs at once.

Is there any reason why it's not possible with other operations like `diff`, `apply`, and `prune`?
I understand that `prune` would be a little tricky to achieve but `diff` and `apply` across multiple tanka envs should be reasonably easy to do for environments that all coexist on the same kubernetes cluster.

Thanks

Simon Let

**Comment by curusarn** [comment]
I do have a use case for `apply` on multiple inline envs that I will gladly describe.

**Comment by Duologic** [comment]
This was a conscious choice, for modifying actions such as `apply` and `prune` it would become fairly complex to handle failure modes so we opted to not implement it. The added benefit did not outweigh the complexity it would introduce.

**Reply by curusarn** [reply]
Hi, thanks for the reply. 

Currently, manually accepting many diffs one after another is a big pain point for us - it's time consuming and error-prone. 
I was hoping that switching to inline Tanka Envs would solve that issue.

Do you have any suggestion how to apply a larger number of environments? We have about a 100 per cluster, regular changes only affect one Tanka env at a time but changes to libraries can cause changes in dozens of them. Btw, we do have some tooling to find what Tanka envs have diffs but we still apply the Tanka envs with diffs individually.

**Reply by Duologic** [reply]
I'd suggest investing into CD, use Tanka to export the manifest and let automated process apply it in each cluster.

**Reply by Duologic** [reply]
Our internal process wraps `tk export` in a script that exports the manifests to a separate git repo, in each cluster we have Flux v1 that continuously pulls and applies that repo.
